src/components/namestable/worker-edit/worker-name-edit-field.components.tsx

```diff
@@ -0,0 +1,79 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+import { Grid, TextField, Typography } from "@material-ui/core";
+import React, { useCallback, useEffect } from "react";
+import { useSelector } from "react-redux";
+import { ApplicationStateModel } from "../../../state/models/application-state.model";
+import { FormFieldErrorLabelStack } from "./form-field-error-label.component";
+import {
+  FormFieldOptions,
+  useFormFieldStyles,
+  WorkerEditComponentMode,
+} from "./worker-edit.models";
+
+export interface WorkerNameEditFieldOptions extends FormFieldOptions {
+  workerName: string;
+  setWorkerName: (newWorkerName: string) => void;
+  mode: WorkerEditComponentMode;
+}
+
+export function WorkerNameEditField({
+  workerName,
+  setWorkerName,
+  setIsFieldValid,
+  mode,
+}: WorkerNameEditFieldOptions): JSX.Element {
+  const classes = useFormFieldStyles();
+
+  const workerNames = useSelector((state: ApplicationStateModel) =>
+    Object.keys(state.actualState.persistentSchedule.present.employee_info.type)
+  );
+
+  const isWorkerWithSameNameExists = useCallback((): boolean => {
+    const isWorkerNameInvalid =
+      workerNames.includes((workerName ?? "").trim()) && mode !== WorkerEditComponentMode.EDIT;
+    return isWorkerNameInvalid;
+  }, [mode, workerName, workerNames]);
+
+  const isWorkerNameEmpty = useCallback((): boolean => {
+    return workerName === "";
+  }, [workerName]);
+
+  useEffect(() => {
+    const isNameValid = !isWorkerNameEmpty() && !isWorkerWithSameNameExists();
+    setIsFieldValid?.(isNameValid);
+  }, [workerName, setIsFieldValid, isWorkerWithSameNameExists, isWorkerNameEmpty]);
+
+  const nameFieldErrorLabels = [
+    {
+      condition: isWorkerWithSameNameExists(),
+      message: `Pracownik o imieniu ${workerName} już istnieje`,
+    },
+    {
+      condition: isWorkerNameEmpty(),
+      message: "Nie zdefiniowano imienia pracownika",
+    },
+  ];
+
+  return (
+    <>
+      <Grid item xs={6}>
+        <Typography className={classes.label}>Imię i nazwisko</Typography>
```

Comment:
Właściwie to nie wiem, czy użycie synekdochy (_pars pro toto_) w tym przypadku jest poprawne. Nie chodzi nam przecież o imię jako samo imię, tylko bardziej o jednoznaczny identyfikator pracownika. To chyba nie jest jakaś niedopuszczalna sytuacja, że w danym miejscu pracują dwie osoby o takich samych imionach.

Reply:
Masz rację. 
Ale nie wiem jak to zamienić: jedyne co przychodzi do głowy - Identyfikator pracownika, ale to brzmi słabo.
Natomiast ta sytuacja o której ty mówisz raczej jest rzadka. 
Moim zdaniem lepiej zostać przy tym co jest teraz .
@prenc, co ty o tym myślisz?

Reply:
Żeby być spójnym można by to nazwać _nazwa pracownika_ (bo zawiera imię i nazwisko, ew. coś jeszcze), ale może to średnio brzmieć.

Co do tego co mówi Agata, to niewiele rozumiem, albo używasz pojęcie imienia wymiennie z nazwą pracownika. Już dopuszczamy kilku pracowników o tym samym imieniu i w teorii, również o tej samej nazwie (case ze zmianą typu umowy).

Reply:
@Ehevi 
Chyba poprawiłem synekdochy xd (jeśli chodziło o to  nazwa pola - to Imię i nazwisko, a w komunikacie piszemy tylko o imieniu)

Co do tego, żeby zamienić Imię i nazwisko na coś innego, to tego już nie robiłem. 
Myśle, że one i tak to zrozumieją w razie czego.

Reply:
To może _Taki pracownik już istnieje_? Skoro zakładamy, że będziemy ich jednoznacznie określać za pomocą imion i nazwisk (co nadal jest synekdochą).

Reply:
@Ehevi 
> Co do tego co mówi Agata, to niewiele rozumiem, albo używasz pojęcie imienia wymiennie z nazwą pracownika. Już dopuszczamy kilku pracowników o tym samym imieniu i w teorii, również o tej samej nazwie (case ze zmianą typu umowy).